pyminiaudio/miniaudio.py

```diff
@@ -894,72 +975,94 @@ def __init__(self, ma_output_format: int = ma_format_s16, nchannels: int = 2,
         _callback_data[id(self)] = self
         self.userdata_ptr = ffi.new("char[]", struct.pack('q', id(self)))
         self._devconfig = lib.ma_device_config_init(lib.ma_device_type_playback)
-        if device_id:
-            _device_id = ffi.addressof(device_id)
-        else:
-            _device_id = ffi.NULL
+        _device_id = pointer_or_null(device_id)
         lib.ma_device_config_set_params(ffi.addressof(self._devconfig), self.sample_rate, self.buffersize_msec,
                                         0, self.format, self.nchannels, 0, 0, _device_id, ffi.NULL)
         self._devconfig.pUserData = self.userdata_ptr
         self._devconfig.dataCallback = lib.internal_data_callback
-        self.audio_producer = None   # type: Optional[AudioProducerType]
+        self.callback_generator = None   # type: Optional[PlaybackCallbackGeneratorType]
```

Comment:
Since this is used in `CaptureDevice` and `DuplexStream` I tried to name it something more generic. Not sure if this is the best name.

Reply:
I think it is okay, it's no longer used for just producing audio, so a new name was needed indeed